### DISCUSSION TITLE
What are "fast" linters?

### DISCUSSION THREAD

**Discussion by jamesrobb** [golangci/golangci-lint#2765] [Q&A] [answered]
Hello,

When looking at the documentation for `golangci-lint run` I see the `--fast` option is document as:

    --fast                           Run only fast linters from enabled linters set (first run won't be fast)

What exactly is a "fast" linter, and what determines that label for a linter? I can certainly understand some are more expensive than others and it might not be desirable to run them all in one's IDE. I'm just looking to satisfy my curiosity on this, and haven't been able to find meaningful documentation on it. Can someone help show me the light?

**Comment by blizzy78** [comment, marked as answer]
There are a few linters that are explicitly declared as being "slow". See [manager.go](https://github.com/golangci/golangci-lint/blob/master/pkg/lint/lintersdb/manager.go) and search for `ConsiderSlow()`.

Also all linters that do type analysis are automatically declared as being slow, which are quite a lot. See [manager.go](https://github.com/golangci/golangci-lint/blob/master/pkg/lint/lintersdb/manager.go) and search for `WithLoadForGoAnalysis()`.
